### ISSUE TITLE
Subscriptions and priority tasks

### ISSUE THREAD

**Issue by thebigmoosey** [OctopusDeploy/Issues#3441]
Certain tasks (like subscriptions) need to have some sort of priority so they are not potentially blocked by long-running deployment tasks that may be maxing out the task cap. Eg. Imagine a task cap of 5, you spin up 5 long-running deployments, suddenly subscription tasks can't run until a space becomes available in the task queue.

## Proposed Solution
We want the advantages of the task system for subscriptions, but without the risk that they'll get blocked/delayed.

We're proposing we add a `HasPriority` property to certain system tasks (like subscriptions) that can be processed by the task queue (even if the task cap has been reached for a given node). This will allow important scheduled tasks like subscriptions to run regularly and not get backed up.

How we go about this is still under discussion. Some options have been proposed:

### Option 1
We could allocate an additional space on top of the current task-cap for priority tasks, so they have space to run (if they are ever blocked).

### Option 2 
We could have a separate task-queue that runs parallel to the existing queue, but have it process only priority tasks.

Will update this issue once we have discussed in more detail.

**Comment by fatmanmclone90** [comment]
Would this also cover normal deployment tasks and being able to set priority based on environment tenant?

e.g. being able to prioritise production environment tasks over test environment tasks etc

**Comment by MichaelJCompton** [comment]
We are proposing to allow one task over the task cap for some special tasks, like subscriptions.  I.E. will operate as per now when the task cap is not reached, but if the the system is running, say, 5 long running deployments with a task cap of 5 this will allow Octopus to tick over essential system tasks.

We are making task scheduling more fair https://github.com/OctopusDeploy/Issues/issues/4168

Beyond that, we are also considering revamping task scheduling to intelligently distribute tasks in HA clusters - so Octopus distributes not just the tasks but the amount of work in tasks across nodes.

**Comment by octoreleasebot** [comment]
Release Note: Each Octopus server node may execute one (non deployment or health check) task beyond it's task cap, so that tasks such as subscriptions are not blocked by long running deployments or health checks.
